### PR TITLE
acl: bootstrap master token from config file

### LINKED ISSUE #7777: Add support for `initial_management` token like in Consul
It would be great if it were possible to preset a token with access-id & secret-id like in consul: https://www.consul.io/docs/agent/options.html#acl_tokens This way nomad could be initialized easier via configuration management tools.

### PR DESCRIPTION
**Pull request by skarrok**
Fixes #7777 
Add support for acl `master_token` like in Consul

### CODE REVIEW

File: nomad/leader.go [hunk 1]
```diff
@@ -318,6 +323,60 @@ func (s *Server) establishLeadership(stopCh chan struct{}) error {
...
+		if err != nil {
+			return err
+		}
+		token, err := state.ACLTokenBySecretID(nil, master)
+		if err != nil {
+			return fmt.Errorf("failed to get master token: %v", err)
+		}
+		if token == nil {
+			token = &structs.ACLToken{
+				AccessorID: uuid.Generate(),
```
Comment (schmichael) [inline nomad/leader.go:345]: Would it be useful to allow the AccessorID to be specified in the HCL as well? Consul doesn't for their `master_token`, but they do allow it for subsequent tokens created via their API in order to ease automation.

**Comment by skarrok** [comment]
Thank you for your review!

Yes, current process is not ideal. We even choose to not use acl in nomad in favor of fully automated idempotent bootstrapping.

I'm sure strong arguments for and against have been made already when this feature landed in consul, so I'm gonna wait for decision.

**Comment by apollo13** [comment]
@schmichael Hi, I have been thinking about your comment and about what would be an easier approach for automation without hard-coding the token in the config file. I only suggested that in #7777 because consul already does it and I like consistency.

What do you think about altering the `/acl/bootstrap` endpoint (or maybe even introduce a new one) to allow the enduser to submit a token pair to it (`AccessorID & SecretID`) which will be used as the initial management token? This way you can safely store the token pair in a system suitable for your automation (for example an ansible vault) but you would not leak it onto the machines.

**Comment by tgross** [comment]
@schmichael what do you think of closing this PR in lieu of the work that @lhaig is doing in https://github.com/hashicorp/nomad/pull/12520? I think that's closer to the suggestion you and @apollo13 had here.

**Comment by schmichael** [comment]
Sounds good to me Tim and @apollo13 gave a :+1: ! Closing this in favor of #12520.

**Pull request closed without merging**